edb/schema/indexes.py

```diff
@@ -302,6 +302,16 @@ def compile_expr_field(
                     singletons=frozenset(singletons),
                 ),
             )
+
+            # Check that the inferred cardinality is no more than 1
+            expr_card = expr.irast.cardinality
+            if expr_card.is_multi():
+                schema_card = expr_card.to_schema_value()[1]
+                raise errors.ResultCardinalityMismatchError(
+                    f'the index expression has cardinality {schema_card}, '
```

Comment:
Please use `possibly more than one element returned by an expression where only singletons are allowed` like we do in all other places where we require singleton expressions.

Reply:
Updated the error message